mne_bids/write.py

```diff
@@ -1254,6 +1263,21 @@ def write_raw_bids(raw, bids_path, events_data=None,
     if not convert and verbose:
         print('Copying data files to %s' % bids_path.fpath.name)
 
+    # If users desire a certain format, will handle auto-conversion
+    if format != 'auto':
+        if format == 'BrainVision' and bids_path.datatype in ['ieeg', 'eeg']:
+            convert = True
+            bids_path.update(extension='.vhdr')
+        elif format == 'FIF' and bids_path.datatype == 'meg':
+            convert = True
+            bids_path.update(extension='.fif')
+        elif format not in CONVERT_FORMATS:
+            raise ValueError(f'The input "format" {format} is not an '
+                             f'accepted input format for `write_raw_bids`. '
+                             f'For example, if you passed in "EDF", mne-bids '
+                             f'does not currently have conversion to EDF. '
+                             f'Please use one of {CONVERT_FORMATS}.')
```

Comment:
@sappelhoff

Reply:
Looks better, but you are not catching the case where format is Brainvision, and Datatype is meg .... Or format is for and Datatype is eeg.

The rest looks good

Reply:
added in new commit.